### PR TITLE
feat(DDI-1059): Added storybook doc for button group

### PR DESCRIPTION
**Pull request by Radha-Alberta**
Added storybook page for Button Group
Added new property "spacing" for button group component

### CODE REVIEW

File: libs/web-components/src/components/button-group/ButtonGroup.svelte [hunk 1]
```diff
@@ -4,12 +4,49 @@
 <script lang="ts">
   // import type { ButtonGroupAlignment } from "@abgov/shared/common";
   // export let alignment: ButtonGroupAlignment = "start";
-  export let gap: "small" | "medium" | "large" = "medium";
-  export let alignment: "start" | "end";
+  //export let gap: "small" | "medium" | "large" = "medium";
```
Comment (chrisolsen) [inline libs/web-components/src/components/button-group/ButtonGroup.svelte:7]: I think we decided to stick with `gap` instead of `spacing`

Comment (Radha-Alberta) [reply libs/web-components/src/components/button-group/ButtonGroup.svelte:7]: Renamed spacing to gap

Comment (chrisolsen) [reply libs/web-components/src/components/button-group/ButtonGroup.svelte:7]: Is there a reason this line, that was added, is commented out?

File: libs/web-components/src/components/button-group/ButtonGroup.svelte [hunk 2]
```diff
@@ -4,12 +4,49 @@
...
+  function isButtonAlignment(value: string): value is ButtonAlignment {
+    return BUTTON_ALIGNMENTS.includes(value);
+  }
+
+  function isGap(value: string): value is Gap {
+    return GAP.includes(value);
+  }
+
+  export let alignment: ButtonAlignment;
+  export let gap: Gap = "relaxed";
```
Comment (chrisolsen) [inline libs/web-components/src/components/button-group/ButtonGroup.svelte:28]: Can you move these lines up to the top with all the other variables. Having them dispersed within the file makes it hard to find them.

File: libs/web-components/src/components/button-group/ButtonGroup.svelte [hunk 3]
```diff
@@ -4,12 +4,49 @@
...
+    if (!isGap(gap)) {
+      console.error("Invalid button group gap");
+    }
+  })
+
 </script>
 
 <!-- HTML -->
-<div style="--alignment: flex-{alignment}; --gap-size: {gap === "small" ? "0.5rem" : "1.25rem"}">
+<div style="--alignment: {_alignment}; --gap-size: {gap === "relaxed" ? "1rem" : "0.75rem"}">
```
Comment (chrisolsen) [inline libs/web-components/src/components/button-group/ButtonGroup.svelte:49]: Why was the `--alignment` changed from `flex-??` to `??`?

File: libs/web-components/src/components/button-group/ButtonGroup.svelte (hunk 1 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/button-group/ButtonGroup.svelte]: Removed the commented lines

File: libs/web-components/src/components/button-group/ButtonGroup.svelte (hunk 2 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/button-group/ButtonGroup.svelte]: Done

**Review by chrisolsen — APPROVED** [review]

**Pull request merged**